tests/Backend/Snowflake/DirectAccessTest.php

```diff
@@ -0,0 +1,93 @@
+{
+    public function testGetDirectAccessCredentials()
+    {
+        $backend = self::BACKEND_SNOWFLAKE;
+        $directAccess = $this->prepareDirectAccess();
+
+        try {
+            $directAccess->getCredentials($backend);
+            $this->fail('Exception should be thrown');
+        } catch (\Keboola\StorageApi\ClientException $e) {
+            $this->assertEquals('storage.directAccess.credentialsForProjectBackendNotFound', $e->getStringCode());
+        }
+
+        try {
+            $directAccess->deleteCredentials($backend);
+            $this->fail('Exception should be thrown');
+        } catch (\Keboola\StorageApi\ClientException $e) {
+            $this->assertEquals('storage.directAccess.tryRemoveNonExistCredentials', $e->getStringCode());
+        }
+
+        try {
+            $directAccess->createCredentials('not-allowed-backend');
+            $this->fail('Exception should be thrown');
+        } catch (\Keboola\StorageApi\ClientException $e) {
+            $this->assertEquals('Invalid request', $e->getMessage());
+            $this->assertEquals('storage.directAccess.validationError', $e->getStringCode());
+        }
+
+        $newCredentials = $directAccess->createCredentials($backend);
+
+        $this->assertArrayHasKey('host', $newCredentials);
+        $this->assertArrayHasKey('username', $newCredentials);
+        $this->assertArrayHasKey('password', $newCredentials);
+
+        $credentials = $directAccess->getCredentials($backend);
+        $this->assertArrayHasKey('username', $credentials);
+        $this->assertSame($newCredentials['username'], $credentials['username']);
+
+        $directAccess->deleteCredentials($backend);
+
+        try {
+            $directAccess->getCredentials($backend);
+            $this->fail('Exception should be thrown');
+        } catch (\Keboola\StorageApi\ClientException $e) {
+            $this->assertEquals('storage.directAccess.credentialsForProjectBackendNotFound', $e->getStringCode());
+        }
+    }
+
+    public function testWithNonAdminToken()
+    {
+        $backend = self::BACKEND_SNOWFLAKE;
+        $newTokenId = $this->_client->createToken(new TokenCreateOptions());
+        $newToken = $this->_client->getToken($newTokenId);
+        $client = new Client([
+            'token' => $newToken['token'],
+            'url' => STORAGE_API_URL,
+        ]);
+
+        $directAccess = new DirectAccess($client);
+
+        try {
+            $directAccess->createCredentials($backend);
+            $this->fail('Exception should be thrown');
+        } catch (\Keboola\StorageApi\ClientException $e) {
+            $this->assertEquals('storage.directAccess.onlyAdminTokenCanCreateCredentials', $e->getStringCode());
+        }
+    }
+
+    /** @return DirectAccess */
+    private function prepareDirectAccess()
+    {
+        $directAccess = new DirectAccess($this->_client);
+
+        try {
+            if ($directAccess->getCredentials(self::BACKEND_SNOWFLAKE)) {
```

Comment:
tady by měl být $backend předaný v parametru. Sice je to teď to samé, ale nebude časem.